prime-router/src/main/kotlin/azure/EmailSenderFunction.kt

```diff
@@ -0,0 +1,184 @@
+var REPORT_STREAM_EMAIL = System.getenv("EMAIL_REPORT_STREAM") ?: "default@noemail.com"
+const val TOS_SUBJECT_BASE = "TOS Agreement for "
+
+/*INFO:
+*  a TemplateID can be found by navigating to our SendGrid dashboard,
+*  expanding the Email API nav list on the left and clicking
+*  Dynamic Templates. The list will show templates with IDs
+*/
+const val TOS_AGREEMENT_TEMPLATE_ID = "d-472779cf554f418a9209acb62d2a48da"
+
+data class TosAgreementForm(
+    val title: String,
+    val firstName: String,
+    val lastName: String,
+    val email: String,
+    val territory: String,
+    val organizationName: String,
+    val operatesInMultipleStates: Boolean,
+    val agreedToTermsOfService: Boolean
+) {
+    fun validate(logger: Logger): Boolean {
+        for (key in TosAgreementForm::class.memberProperties) {
+            val value = key.get(this)
+            when (key.name) {
+                "title" -> if (!verifyNotExceededLimit(key, value as String, logger)) return false
+                "firstName" -> if (!verifyBoth(key, value as String, logger)) return false
+                "lastName" -> if (!verifyBoth(key, value as String, logger)) return false
+                "email" -> if (!verifyBoth(key, value as String, logger)) return false
+                "territory" -> if (!verifyBoth(key, value as String, logger)) return false
+                "organizationName" -> if (!verifyBoth(key, value as String, logger)) return false
+                "agreedToTermsOfService" -> if (!verifyAgreed(logger)) return false
+            }
+        }
+        return verifyAgreed(logger)
+    }
+
+    private fun verifyAgreed(logger: Logger): Boolean {
+        if (!this.agreedToTermsOfService) logger.info("Uh oh, your agreement to the Terms of Service is marked false")
+        return this.agreedToTermsOfService
+    }
+
+    private fun verifyBoth(key: KProperty1<TosAgreementForm, *>, value: String, logger: Logger): Boolean {
+        if (!verifyIsNotBlank(key, value, logger) ||
+            !verifyNotExceededLimit(key, value, logger)
+        ) return false
+        return true
+    }
+
+    private fun verifyIsNotBlank(key: KProperty1<TosAgreementForm, *>, value: String, logger: Logger): Boolean {
+        if (value.isBlank()) logger.info("Uh oh, \"${key.name}\" in your request body is Null")
+        return value.isNotBlank()
+    }
+
+    private fun verifyNotExceededLimit(key: KProperty1<TosAgreementForm, *>, value: String, logger: Logger): Boolean {
+        if (value.length > 255) logger.info("Uh oh, \"${key.name}\" has exceeded the character limit")
+        return value.length <= 255
+    }
+}
+
+class EmailSenderFunction {
+
+    @FunctionName("emailRegisteredOrganization")
+    @StorageAccount("AzureWebJobsStorage")
+    fun emailRegisteredOrganization(
+        @HttpTrigger(
+            name = "emailRegisteredOrganization",
+            methods = [HttpMethod.POST],
+            authLevel = AuthorizationLevel.ANONYMOUS,
```

Comment:
We need some restriction on this endpoint. Either auth if that's available when it's being used, or can we restrict it somehow to only allow the frontend website to use it?

Reply:
Added! Check the latest updates 👍🏼